modules/saml/www/sp/saml2-acs.php

```diff
@@ -186,6 +191,7 @@
         $foundAuthnStatement = true;
     }
 }
+$assertion = end($assertions);
```

Comment:
Why is this necessary? Is psalm complaining that `$assertion` is no longer defined outside of the loop's scope?

Reply:
Exactly; if `$assertions` would be empty, `$assertion` would never be defined.